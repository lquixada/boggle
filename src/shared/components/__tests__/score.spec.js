import React from 'react'
import { mount } from 'enzyme'
import { fromJS } from 'immutable'

import { Score } from '..'

describe('<Score />', () => {
  it('is empty by default', () => {
    const component = mount(<Score attempts={fromJS([])} />)

    expect(component.find('tr').exists()).toBe(false)
  })

  it('adds a valid attempt', () => {
    const attempts = fromJS([{ word: 'someword', score: '8' }])
    const component = mount(<Score attempts={attempts} />)

    expect(component).toMatchSnapshot()
  })

  it('adds an invalid attempt', () => {
    const attempts = fromJS([{ word: 'someword', score: 'âœ˜' }])
    const component = mount(<Score attempts={attempts} />)

    expect(component).toMatchSnapshot()
  })

  it('shows the score', () => {
    const component = mount(<Score attempts={fromJS([])} counter={8} />)

    expect(component).toMatchSnapshot()
  })
})
