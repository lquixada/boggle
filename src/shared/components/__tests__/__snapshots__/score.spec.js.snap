// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Score /> adds a valid attempt 1`] = `
<Score
  attempts={
    Immutable.List [
      Immutable.Map {
        "word": "someword",
        "score": "8",
      },
    ]
  }
>
  <styled.div>
    <div
      className="sc-bdVaJa eJrItC"
    >
      <styled.header>
        <header
          className="sc-bwzfXH cIaGHc"
        >
          <h2>
            Score: 
          </h2>
        </header>
      </styled.header>
      <styled.section>
        <section
          className="sc-htpNat bbFALC"
        >
          <styled.table>
            <table
              className="sc-bxivhb cYLiMQ"
            >
              <tbody>
                <Attempt
                  attempt={
                    Immutable.Map {
                      "word": "someword",
                      "score": "8",
                    }
                  }
                  key="0"
                >
                  <tr>
                    <styled.td>
                      <td
                        className="sc-ifAKCX fSZhoQ"
                      >
                        someword
                      </td>
                    </styled.td>
                    <styled.td>
                      <td
                        className="sc-ifAKCX fSZhoQ"
                      >
                        8
                      </td>
                    </styled.td>
                  </tr>
                </Attempt>
              </tbody>
            </table>
          </styled.table>
        </section>
      </styled.section>
    </div>
  </styled.div>
</Score>
`;

exports[`<Score /> adds an invalid attempt 1`] = `
<Score
  attempts={
    Immutable.List [
      Immutable.Map {
        "word": "someword",
        "score": "✘",
      },
    ]
  }
>
  <styled.div>
    <div
      className="sc-bdVaJa eJrItC"
    >
      <styled.header>
        <header
          className="sc-bwzfXH cIaGHc"
        >
          <h2>
            Score: 
          </h2>
        </header>
      </styled.header>
      <styled.section>
        <section
          className="sc-htpNat bbFALC"
        >
          <styled.table>
            <table
              className="sc-bxivhb cYLiMQ"
            >
              <tbody>
                <Attempt
                  attempt={
                    Immutable.Map {
                      "word": "someword",
                      "score": "✘",
                    }
                  }
                  key="0"
                >
                  <tr>
                    <styled.td>
                      <td
                        className="sc-ifAKCX fSZhoQ"
                      >
                        someword
                      </td>
                    </styled.td>
                    <styled.td>
                      <td
                        className="sc-ifAKCX fSZhoQ"
                      >
                        ✘
                      </td>
                    </styled.td>
                  </tr>
                </Attempt>
              </tbody>
            </table>
          </styled.table>
        </section>
      </styled.section>
    </div>
  </styled.div>
</Score>
`;

exports[`<Score /> shows the score 1`] = `
<Score
  attempts={Immutable.List []}
  counter={8}
>
  <styled.div>
    <div
      className="sc-bdVaJa eJrItC"
    >
      <styled.header>
        <header
          className="sc-bwzfXH cIaGHc"
        >
          <h2>
            Score: 
            8
          </h2>
        </header>
      </styled.header>
      <styled.section>
        <section
          className="sc-htpNat bbFALC"
        >
          <styled.table>
            <table
              className="sc-bxivhb cYLiMQ"
            >
              <tbody>
                <undefined />
              </tbody>
            </table>
          </styled.table>
        </section>
      </styled.section>
    </div>
  </styled.div>
</Score>
`;
